app/gui2/src/App.vue

```diff
@@ -1,46 +1,76 @@
 <script setup lang="ts">
 import HelpScreen from '@/components/HelpScreen.vue'
 import { provideAppClassSet } from '@/providers/appClass'
+import { provideEventLogger } from '@/providers/eventLogging'
 import { provideGuiConfig } from '@/providers/guiConfig'
-import { useProjectStore } from '@/stores/project'
-import { useSuggestionDbStore } from '@/stores/suggestionDatabase'
 import { registerAutoBlurHandler } from '@/util/autoBlur'
-import { configValue, type ApplicationConfig, type ApplicationConfigValue } from '@/util/config'
+import {
+  baseConfig,
+  configValue,
+  mergeConfig,
+  type ApplicationConfigValue,
+  type StringConfig,
+} from '@/util/config'
 import ProjectView from '@/views/ProjectView.vue'
-import { isDevMode } from 'shared/util/detect'
-import { computed, onMounted, onUnmounted, toRaw } from 'vue'
+import { useEventListener } from '@vueuse/core'
+import { computed, toRef, watch } from 'vue'
+import { initializePrefixes } from './util/ast/node'
+import { urlParams } from './util/urlParams'
 
 const props = defineProps<{
-  config: ApplicationConfig
-  accessToken: string | null
-  unrecognizedOptions: string[]
+  config: StringConfig
+  projectId: string
+  logEvent: LogEvent
+  hidden: boolean
+  ignoreParamsRegex?: RegExp
 }>()
 
 const classSet = provideAppClassSet()
 
-provideGuiConfig(computed((): ApplicationConfigValue => configValue(props.config)))
+initializePrefixes()
 
-registerAutoBlurHandler()
+const logger = provideEventLogger(toRef(props, 'logEvent'), toRef(props, 'projectId'))
+watch(
+  [toRef(props, 'projectId')],
```

Comment:
i assume `props.projectId` comes from React so you wouldn't get reactivity. although, i also assume `toRef(props, 'projectId')` without the wrapping array would be valid, since it *is* a ref (i.e. it is a valid watch source)?

Reply:
The array is indeed unnecessary, but `toRef` is needed to pass a ref the watcher, not the current value.